workflow-manager/main.go

```diff
@@ -268,50 +666,41 @@ func launch(ctx context.Context, readyBatches []string, ageLimit time.Duration)
 
 	log.Printf("starting %d jobs", len(readyBatches))
 	for _, batchName := range readyBatches {
-		if err := startJob(ctx, clientset, batchName, ageLimit); err != nil {
+		if err := startIntakeJob(ctx, clientset, batchName, ageLimit); err != nil {
 			return fmt.Errorf("starting job for batch %q: %w", batchName, err)
 		}
 	}
 	return nil
 }
 
-func startJob(
+func startIntakeJob(
 	ctx context.Context,
 	clientset *kubernetes.Clientset,
 	batchName string,
 	ageLimit time.Duration,
 ) error {
-	// batchName is like "kittens-seen/2020/10/31/20/29/b8a5579a-f984-460a-a42d-2813cbf57771"
-	pathComponents := strings.Split(batchName, "/")
-	batchID := pathComponents[len(pathComponents)-1]
-	aggregationID := pathComponents[0]
-	batchDate := pathComponents[1 : len(pathComponents)-1]
-
-	if len(batchDate) != 5 {
-		return fmt.Errorf("malformed date in %q. Expected 5 date components, got %d", batchName, len(batchDate))
+	batchPath, err := newBatchPath(batchName)
+	if err != nil {
+		return err
 	}
-	var dateComponents []int
-	for _, c := range batchDate {
-		parsed, err := strconv.ParseInt(c, 10, 64)
-		if err != nil {
-			return fmt.Errorf("parsing date component %q in %q: %w", c, batchName, err)
-		}
-		dateComponents = append(dateComponents, int(parsed))
+	batchDate, err := batchPath.batchDateTime()
```

Comment:
Go style would be to omit the redundant `batch` in the naming of this function. And `DateTime` is a little redundant since we're just returning a `time`. So I'd call this method `.time()`.

Reply:
I renamed a bunch of members and functions to remove redundant `batch` occurences.